src/main/java/com/clearspring/analytics/stream/cardinality/HyperLogLogPlus.java

```diff
@@ -789,11 +805,11 @@ public void addAll(HyperLogLogPlus other) throws HyperLogLogPlusMergeException {
 
         if ((format == Format.SPARSE) && (other.format == Format.SPARSE)) {
             sparseSet = mergeEstimators(other);
-            // Convert to normal mode if needed. 
-            // Since offer trigger the switch to the normal mode on only when 
-            // the tmpSet is full and the threshold is reached, we follow the same 
+            // Convert to normal mode if needed.
+            // Since offer trigger the switch to the normal mode on only when
+            // the tmpSet is full and the threshold is reached, we follow the same
             // behavior here to ease testing
-            if (sparseSet.length > (sparseSetThreshold + sortThreshold)) {
+            if (sparseSet.length > (sparseSetThreshold * (SPARSE_SET_TEMP_SET_RATIO + 1)) / SPARSE_SET_TEMP_SET_RATIO) {
```

Comment:
Sure. There's a unit test TestHyperLogLogPlus#testMergeSelf() that merges a cardinality estimator with itself and verifies that the cardinality estimates are identical. To guarantee that they are identical the original implementation used  if (sparseSet.length > (sparseSetThreshold + sortThreshold)) which expands to sparseSet.length > (sparseSetThreshold + sparseSetThreshold / 4). I didn't know how to correctly update the unit test. If we change the unit test then we can change the statement to (sparseSet.length > sparseSetThreshold). If we don't change the unit test then commit will have the new behavior be identical to its original behavior (the default value of SPARSE_SET_TEMP_SET_RATIO is 4).